flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java

```diff
@@ -0,0 +1,277 @@
+import org.apache.flink.table.expressions.FieldReferenceExpression;
+import org.apache.flink.table.expressions.UnresolvedCallExpression;
+import org.apache.flink.table.expressions.UnresolvedReferenceExpression;
+import org.apache.flink.table.expressions.ValueLiteralExpression;
+import org.apache.flink.table.expressions.utils.ApiExpressionDefaultVisitor;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.iceberg.expressions.And;
+import org.apache.iceberg.expressions.Not;
+import org.apache.iceberg.expressions.Or;
+import org.apache.iceberg.expressions.UnboundPredicate;
+import org.apache.iceberg.util.DateTimeUtil;
+import org.junit.Test;
+
+import static org.apache.flink.table.api.Expressions.$;
+import static org.apache.flink.table.api.Expressions.lit;
+import static org.junit.Assert.assertEquals;
+
+public class TestFlinkFilters {
+
+  private static final TableSchema TABLE_SCHEMA = TableSchema.builder()
+      .field("field1", DataTypes.INT())
+      .field("field2", DataTypes.BIGINT())
+      .field("field3", DataTypes.FLOAT())
+      .field("field4", DataTypes.DOUBLE())
+      .field("field5", DataTypes.STRING())
+      .field("field6", DataTypes.BOOLEAN())
+      .field("field7", DataTypes.BINARY(10))
+      .field("field8", DataTypes.DECIMAL(10, 2))
+      .field("field9", DataTypes.DATE())
+      .field("field10", DataTypes.TIME())
+      .field("field11", DataTypes.TIMESTAMP())
+      .field("field12", DataTypes.TIMESTAMP_WITH_TIME_ZONE())
+      .field("field13", DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE())
+      .build();
+
+  @Test
+  public void testFlinkDataTypeEqual() {
+    matchLiteral("field1", 1, 1);
+    matchLiteral("field2", 10L, 10L);
+    matchLiteral("field3", 1.2F, 1.2F);
+    matchLiteral("field4", 3.4D, 3.4D);
+    matchLiteral("field5", "abcd", "abcd");
+    matchLiteral("field6", true, true);
+    matchLiteral("field7", new byte[] {'a', 'b'}, new byte[] {'a', 'b'});
+    matchLiteral("field8", BigDecimal.valueOf(10), BigDecimal.valueOf(10));
+
+    LocalDate date = LocalDate.parse("2020-12-23");
+    matchLiteral("field9", date, DateTimeUtil.daysFromDate(date));
+
+    LocalTime time = LocalTime.parse("12:13:14");
+    matchLiteral("field10", time, DateTimeUtil.microsFromTime(time));
+
+    LocalDateTime dateTime = LocalDateTime.parse("2020-12-23T12:13:14");
+    matchLiteral("field11", dateTime, DateTimeUtil.microsFromTimestamp(dateTime));
+
+    Instant instant = Instant.parse("2020-12-23T12:13:14.00Z");
+    matchLiteral("field12", instant, DateTimeUtil.microsFromInstant(instant));
+
+    matchLiteral("field13", instant, DateTimeUtil.microsFromInstant(instant));
+  }
+
+  @Test
+  public void testEquals() {
+    UnboundPredicate<Integer> expected = org.apache.iceberg.expressions.Expressions.equal("field1", 1);
+
+    org.apache.iceberg.expressions.Expression actual =
+        FlinkFilters.convert(resolve(Expressions.$("field1").isEqual(Expressions.lit(1)))).get();
```

Comment:
Nit: better to check `ifPresent` before `get` the value from `Optional`.

Reply:
I changed all `get` to `orElse(null)`, my initial thought was that in this specified case, the result should not return null, so I did not add judgment